packages/polymath-ui/src/components/EthNetworkWrapper/EthNetworkWrapper.js

```diff
@@ -56,14 +54,34 @@ export class EthNetworkWrapper extends Component<Props> {
     }
   }
 
+  init = () => {
+    this.props.init(this.props.networks);
+  };
+
+  handleAuthRequest() {
+    this.props.requestAuthorization();
+  }
+
   render() {
-    if (this.props.isConnected) {
-      return this.props.children;
+    const {
+      isConnected,
+      isFailed,
+      Loading,
+      errorRender,
+      error,
+      children,
+    } = this.props;
+
+    if (isFailed) {
+      return errorRender({
+        networkError: error,
+        onRequestAuth: this.handleAuthRequest.bind(this),
+      });
     }
-    if (!this.props.isFailed) {
-      return this.props.loading;
+    if (!isFailed && !isConnected) {
```

Comment:
We can remove the `!isFailed` here
```suggestion
    if (!isConnected) {
```